lib/Jsrt/Jsrt.cpp

```diff
@@ -5329,4 +5329,45 @@ CHAKRA_API JsSetHostPromiseRejectionTracker(_In_ JsHostPromiseRejectionTrackerCa
     /*allowInObjectBeforeCollectCallback*/true);
 }
 
+CHAKRA_API JsGetProxyProperties (_In_ JsValueRef object, _Out_ bool* isProxy, _Out_opt_ JsValueRef* target, _Out_opt_ JsValueRef* handler)
+{
+    return ContextAPINoScriptWrapper_NoRecord([&](Js::ScriptContext * scriptContext) -> JsErrorCode {
+        VALIDATE_INCOMING_REFERENCE(object, scriptContext);
+        PARAM_NOT_NULL(isProxy);
+
+        *isProxy = Js::JavascriptProxy::Is(object);
+
+        if (!*isProxy)
+        {
+            return JsNoError;
```

Comment:
Unless I'm mistaken, `_Out_opt_` has the same semantics as `_Out_` when it comes to initialization. That means this function should not return without initializing `target` and `handler` if they are not null. We should just initialize both of them to `JS_INVALID_REFERENCE` before the assignment to `isProxy` if they're non-null.

Reply:
Yeah, `_Out_opt` from what I've been able to gather just means the out pointer itself can be nullptr - if it's a non-null pointer then it should indeed be treated the same as an `_Out_` parameter.